public/search/sql_async_search_interceptor.ts

```diff
@@ -77,18 +87,29 @@ export class SQLAsyncSearchInterceptor extends SearchInterceptor {
     };
 
     const onPollingSuccess = (pollingResult: any) => {
-      if (pollingResult && pollingResult.body.meta.status === 'SUCCESS') {
-        return false;
-      }
-      if (pollingResult && pollingResult.body.meta.status === 'FAILED') {
-        const jsError = new Error(pollingResult.data.error.response);
-        this.deps.toasts.addError(jsError, {
-          title: i18n.translate('queryEnhancements.sqlQueryError', {
-            defaultMessage: 'Could not complete the SQL async query',
-          }),
-          toastMessage: pollingResult.data.error.response,
-        });
-        return false;
+      if (pollingResult) {
+        const statusStr: string = (pollingResult.body.meta.status as string).toUpperCase();
+        const status = SparkJobState[statusStr as keyof typeof SparkJobState];
+        switch (status) {
+          case SparkJobState.SUCCESS:
+            return false;
+          case SparkJobState.FAILED:
+            const jsError = new Error(pollingResult.data.error.response);
+            this.deps.toasts.addError(jsError, {
+              title: i18n.translate('queryEnhancements.sqlQueryError', {
+                defaultMessage: 'Could not complete the SQL async query',
+              }),
+              toastMessage: pollingResult.data.error.response,
+            });
+            return false;
+          default:
+            if (request.params?.progress_query_id) {
+              this.uiActions.getTrigger(ASYNC_TRIGGER_ID).exec({
+                query_id: request.params.progress_query_id,
+                query_status: status,
+              });
+            }
```

Comment:
Core logic is here: emit a UI action trigger for in-progress states. Rest of the PR is plumbing.